rfcs/0000-ci-all-nix-prs.md

```diff
@@ -0,0 +1,60 @@
+---
+feature: ci-all-nix-prs
+start-date: 2019-10-30
+author: John Ericson (@Ericson2314)
+co-authors: (find a buddy later to help our with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Build all Nix PRs in CI.
+Do not merge any PR until it passes CI.
+
+# Motivation
+[motivation]: #motivation
+
+There is a (famous blog post)[blog-post] that everyone is sloppy and doing CI wrong.
+This isn't just bad for releasing software smoothly, but also increases the burden for new contributors because it is harder to judge the correctness of PRs at a glance (is it broken? Did I break it?).
+I personally find it harder to contribute, I have to worry about double checking all my work on platforms I don't have as-easy access to, like Darwin.
+
+We cannot yet do this for all of Nixpkgs, sadly, due to resource limits.
+But, there is no reason we cannot do it for Nix itself.
+
+# Detailed design
+[design]: #detailed-design
+
+Set up Hydra declarative jobsets to build all Nix PRs.
```

Comment:
Having used these in the past, they rarely work well. I think the Hydra model only really works for "post-merge" CI, not "pre-merge" CI. A tool like OfBorg (or something else) seems like a good solution to "pre-merge" CI.